src/bin.js

```diff
@@ -2,29 +2,124 @@
 
 'use strict'
 
-const signalling = require('./index')
-const argv = require('minimist')(process.argv.slice(2))
+/* eslint-disable no-console */
 
-let server
+const Server = require('./')
+const Libp2p = require('libp2p')
 
-/* eslint-disable no-console */
+const WS = require('libp2p-websockets')
+const TCP = require('libp2p-tcp')
+const MULTIPLEX = require('libp2p-multiplex')
+const SPDY = require('libp2p-spdy')
+const SECIO = require('libp2p-secio')
+
+const inDocker = Boolean(process.env.DOCKERMODE)
```

Comment:
This shouldn't be a concern to the application. Seems to only be used to set the `config` directory so instead we should accept a environment variable for setting that path.

Reply:
The env variable for that already exists. It's more of a convenience thing so the user does not have to specify `-e CONFIG=/data/config.json` every time when running the container

Reply:
Oh wait... I could actually embed _that_ in the Dockerfile *facepalm of doom* ;)